src/platform/OpenThread/GenericNetworkCommissioningThreadDriver.cpp

```diff
@@ -175,18 +175,25 @@ void GenericThreadDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * c
     }
 }
 
-void GenericThreadDriver::ScanNetworks(ThreadDriver::ScanCallback * callback)
+static void ScanNetworksNow(System::Layer *, void * callbackPtr)
 {
-    CHIP_ERROR err = DeviceLayer::ThreadStackMgrImpl().StartThreadScan(callback);
-    if (err != CHIP_NO_ERROR)
+    ThreadDriver::ScanCallback * callback = static_cast<ThreadDriver::ScanCallback *>(callbackPtr);
+
+    if (DeviceLayer::ThreadStackMgrImpl().StartThreadScan(callback) != CHIP_NO_ERROR)
     {
-        mScanStatus.SetValue(Status::kUnknownError);
         callback->OnFinished(Status::kUnknownError, CharSpan(), nullptr);
     }
-    else
+}
+
+void GenericThreadDriver::ScanNetworks(ThreadDriver::ScanCallback * callback)
+{
+    using namespace System::Clock::Literals;
+
+    // While the Thread network discovery is happening, the communication over Thread can be disrupted,
+    // so introduce a delay to help deliver an ACK for the ScanNetwork command before the discovery begins.
+    if (DeviceLayer::SystemLayer().StartTimer(CHIP_DEVICE_CONFIG_THREAD_SCAN_DELAY, ScanNetworksNow, callback) != CHIP_NO_ERROR)
```

Comment:
Should we just manually send a standalone ack in the cluster like we do for other commands that might not send an ack expeditiously enough and have a shorter delay here instead of aiming for a delay that's longer than the MRP standalone ack timeout?  Since we know we don't plan to send a response before the standalone ack goes out, why make it wait?

Reply:
Good idea! I just didn't know we have an API for that.